af/fastpath/fastpath/core.py

```diff
@@ -0,0 +1,574 @@
+
+# Feeds measurements from Collectors over SSH
+import fastpath.sshfeeder as sshfeeder
+
+# Feeds measurements from S3
+import fastpath.s3feeder as s3feeder
+
+# Push measurements into Postgres
+import fastpath.db as db
+
+from fastpath.metrics import setup_metrics
+
+import fastpath.utils
+
+LOCALITY_VALS = ("general", "global", "country", "isp", "local")
+
+NUM_WORKERS = 15
+
+log = logging.getLogger("fastpath")
+metrics = setup_metrics(name="fastpath")
+
+conf = Namespace()
+fingerprints = None
+
+
+def parse_date(d):
+    return datetime.strptime(d, "%Y-%m-%d").date()
+
+
+def setup():
+    os.environ["TZ"] = "UTC"
+    global conf
+    ap = ArgumentParser(__doc__)
+    ap.add_argument("--start-day", type=lambda d: parse_date(d))
+    ap.add_argument("--end-day", type=lambda d: parse_date(d))
+    ap.add_argument("--devel", action="store_true", help="Devel mode")
+    ap.add_argument(
+        "--update",
+        action="store_true",
+        help="Update summaries and files instead of logging an error",
+    )
+    ap.add_argument("--interact", action="store_true", help="Interactive mode")
+    ap.add_argument(
+        "--stop-after", type=int, help="Stop after feeding N measurements", default=None
+    )
+    conf = ap.parse_args()
+    if conf.devel:
+        root = Path(os.getcwd())
+        format = "%(relativeCreated)d %(process)d %(levelname)s %(name)s %(message)s"
+        logging.basicConfig(stream=sys.stdout, level=logging.DEBUG, format=format)
+    else:
+        root = Path("/")
+        log.addHandler(JournalHandler(SYSLOG_IDENTIFIER="fastpath"))
+        log.setLevel(logging.DEBUG)
+
+    conf.conffile = root / "etc/fastpath.conf"
+    log.info("Using conf file %r", conf.conffile)
+    conf.vardir = root / "var/lib/fastpath"
+    conf.cachedir = conf.vardir / "cache"
+    conf.s3cachedir = conf.cachedir / "s3"
+    conf.dfdir = conf.vardir / "dataframes"
+    conf.outdir = conf.vardir / "output"
+    conf.msmtdir = conf.outdir / "measurements"
+    for p in (
+        conf.vardir,
+        conf.cachedir,
+        conf.s3cachedir,
+        conf.dfdir,
+        conf.outdir,
+        conf.msmtdir,
+    ):
+        p.mkdir(parents=True, exist_ok=True)
+
+
+def per_s(name, item_count, t0):
+    """Generate a gauge metric of items per second
+    """
+    delta = time.time() - t0
+    if delta > 0:
+        metrics.gauge(f"{name}_per_s", item_count / delta)
+
+
+@metrics.timer("clean_caches")
+def clean_caches():
+    """Cleanup local caches.
+    """
+    # Access times are updated on file load.
+    # FIXME: use cache locations correctly
+    now = time.time()
+    threshold = 3600 * 24 * 3
+    for f in conf.s3cachedir.iterdir():
+        if not f.is_file():
+            continue
+        age_s = now - f.stat().st_atime
+        if age_s > threshold:
+            log.debug("Deleting %s", f)
+            metrics.gauge("deleted_cache_file_age", age_s)
+            # TODO: delete
+
+
+expected_colnames = {
```

Comment:
I don't see where this variable is being used.

Reply:
It's not used at the moment. Added comment.